src/librustdoc/html/render/mod.rs

```diff
@@ -1634,6 +1634,18 @@ fn render_impl(
         }
 
         if let Some(ref dox) = cx.shared.maybe_collapsed_doc_value(&i.impl_item) {
+            if render_mode == RenderMode::Normal
```

Comment:
Hmm, shouldn't it be impossible for `render_mode` to be `RenderMode::Deref` and for `trait.is_some()` to be true at the same time, since the recursive deref check only documents inherent impls? Why check both?

I wonder if we could combine those somehow maybe? Or maybe it makes more sense to show traits that the deref target implements :thinking:

Reply:
It's already checked just a few lines above so completely unneeded, you're absolutely right.

Reply:
As for showing the traits on the Deref types, I think it would just be too much information (or we will need to think how to display them). But that's for another time. :smile: Maybe open an issue so we can talk about it later on?